llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1015,6 +1017,81 @@ void SpirvLowerGlobal::lowerInOutInPlace() {
   }
 }
 
+// =====================================================================================================================
+// @param builtIn : BuiltIn value
+// @param elemIdx : Element Index of struct
+Value *SpirvLowerGlobal::createRaytracingBuiltIn(BuiltIn builtIn, Value *elemIdx) {
+  Value *builtinValue = nullptr;
+  switch (builtIn) {
+  case BuiltInLaunchIdKHR:
+    builtinValue = m_builder->create<DispatchRaysIndexOp>();
+    break;
+  case BuiltInLaunchSizeKHR:
+    builtinValue = m_builder->create<DispatchRaysDimensionsOp>();
+    break;
+  case BuiltInWorldRayOriginKHR:
+    builtinValue = m_builder->create<WorldRayOriginOp>();
+    break;
+  case BuiltInWorldRayDirectionKHR:
+    builtinValue = m_builder->create<WorldRayDirectionOp>();
+    break;
+  case BuiltInObjectRayOriginKHR:
+    builtinValue = m_builder->create<ObjectRayOriginOp>();
+    break;
+  case BuiltInObjectRayDirectionKHR:
+    builtinValue = m_builder->create<ObjectRayDirectionOp>();
+    break;
+  case BuiltInRayTminKHR:
+    builtinValue = m_builder->create<RayTminOp>();
+    break;
+  case BuiltInRayTmaxKHR:
+    builtinValue = m_builder->create<RayTcurrentOp>();
+    break;
+  case BuiltInInstanceCustomIndexKHR:
+    builtinValue = m_builder->create<InstanceIndexOp>();
+    break;
+  case BuiltInObjectToWorldKHR:
+    builtinValue = m_builder->create<ObjectToWorldOp>();
+    break;
+  case BuiltInWorldToObjectKHR:
+    builtinValue = m_builder->create<WorldToObjectOp>();
+    break;
+  case BuiltInHitKindKHR:
+    builtinValue = m_builder->create<HitKindOp>();
+    break;
+  case BuiltInHitTriangleVertexPositionsKHR:
+    builtinValue = m_builder->create<TriangleVertexPositionsOp>(elemIdx);
```

Comment:
what another operation do you mean?

Reply:
Sorry, what I meant, was: Since the list of dialect operations built in this function is probably not complete (see Nicolai's comment) - will `elemIdx` be used in any other case except `TriangleVertexPositionsOp`? It's a bit unfortunate that the argument `elemIdx` is only used in a very small part of `createRaytracingBuiltIn`.